src/actions/channels.ts

```diff
@@ -724,6 +724,36 @@ export function deleteChannel(channelId: string): ActionFunc {
     };
 }
 
+export function unarchiveChannel(channelId: string): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let state = getState();
+        const viewArchivedChannels = state.entities.general.config.ExperimentalViewArchivedChannels === 'true';
+
+        try {
+            await Client4.unarchiveChannel(channelId);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        state = getState();
+        const {currentChannelId} = state.entities.channels;
+        if (channelId === currentChannelId && !viewArchivedChannels) {
+            const teamId = getCurrentTeamId(state);
+            const channelsInTeam = getChannelsNameMapInTeam(state, teamId);
+            const channel = getChannelByName(channelsInTeam, getRedirectChannelNameForTeam(state, teamId));
+            if (channel && channel.id) {
+                dispatch({type: ChannelTypes.SELECT_CHANNEL, data: channel.id}, getState);
```

Comment:
Also, `getState` doesn't need to be passed into `dispatch`. Anywhere we do that is actually unnecessary.
```suggestion
                dispatch({type: ChannelTypes.SELECT_CHANNEL, data: channel.id});
```